### PR TITLE
Fixes #28056: Use ForemanModal in Katello modals

### PR DESCRIPTION
**Pull request by jeremylenz**
Converting some existing Katello modals to the new ForemanModal component.

#### Depends on:
~~* [RM #27853 - Integrate modals with Redux store] [merged] (https://github.com/theforeman/foreman/pull/7105)~~

#### To test this PR:
~* IMPORTANT: In Foreman repo, check out branch `jeremylenz/27853-modals-redux`~
* ~Make sure your Foreman `package.json` lists `"@theforeman/vendor": "^2.15.7"`~
* In Foreman repo, run `npm install`

### CODE REVIEW

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js [hunk 1]
```diff
@@ -52,10 +57,16 @@ class ManageManifestModal extends Component {
   }
 
   hideModal = () => {
-    this.setState({ showModal: false, showDeleteManifestModalDialog: false });
-    this.props.onClose();
+    this.hideDeleteManifestModal();
+    this.props.setModalClosed({ id: MANAGE_MANIFEST_MODAL_ID });
```
Comment (MariaAga) [inline webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: `hideDeleteManifestModal` is a function that calls `setModalClosed`. So in `hideModal` `setModalClosed` is called twice, why?

Comment (jeremylenz) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: @MariaAga There are two modals on this page that I converted to ForemanModal-- the `ManageManifestModal` itself and also the `DeleteManifestModalDialog` which is the confirmation when you delete a manifest.  So, `setModalClosed` is called once for each modal.  Notice the id passed to `setModalClosed` is different in each call.

It's equivalent to 
```javascript
this.props.setModalClosed({ id: DELETE_MANIFEST_MODAL_ID });
this.props.setModalClosed({ id: MANAGE_MANIFEST_MODAL_ID });
```

**Comment by jeremylenz** [comment]
Leaving [WIP] until Foreman #7105 is merged, but can be reviewed now by checking out that branch.

**Comment by jeremylenz** [comment]
Katello test failures are expected until Foreman [#7105](https://github.com/theforeman/foreman/pull/7105) is merged

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js (hunk 1 above)
Comment (MariaAga) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: I didn't notice thanks for clearing this up!

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js [hunk 2]
```diff
@@ -150,166 +153,155 @@ class ManageManifestModal extends Component {
     };
 
     return (
-      <Modal show={this.state.showModal} onHide={this.hideModal}>
-        <Modal.Header>
+      <ForemanModal id={MANAGE_MANIFEST_MODAL_ID} title="Manage Manifest">
```
Comment (MariaAga) [inline webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: The Model doesn't show the title.
Also the title should be translated

Comment (jeremylenz) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: @MariaAga Thanks, good catch :)

I updated the Foreman PR to fix the title not showing, and I added translate here.

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js [hunk 3]
```diff
@@ -150,166 +153,147 @@ class ManageManifestModal extends Component {
...
+              <Table
+                rows={manifestHistory.results}
+                columns={columns}
+                emptyState={emptyStateData()}
+              />
+            </LoadingState>
+          </Tab>
+        </Tabs>
+        <ForemanModal.Footer>
           <Button bsStyle="primary" onClick={this.hideModal}>
```
Comment (MariaAga) [inline webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: Why should it hide both modals if this button is only accessible from the `Manage Manifest` ?

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js [hunk 4]
```diff
@@ -82,13 +91,7 @@ class ManageManifestModal extends Component {
...
     this.setState({ actionInProgress: true });
     this.props.delete();
-    this.showDeleteManifestModal(false);
-  };
-
-  showDeleteManifestModal = (show) => {
-    this.setState({
-      showDeleteManifestModalDialog: show,
-    });
+    this.hideDeleteManifestModal();
```
Comment (MariaAga) [inline webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: hideModal already calls  hideDeleteManifestModal

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js (hunk 3 above)
Comment (jeremylenz) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: @MariaAga It was just for peace of mind mostly.  Wanted to guarantee that there would never be a state where the delete confirmation dialog is showing but the ManageManifestModal is not.

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js (hunk 4 above)
Comment (jeremylenz) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: @MariaAga true, updating

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js (hunk 3 above)
Comment (MariaAga) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: Makes sense

File: webpack/scenes/Subscriptions/SubscriptionsPage.js [hunk 5]
```diff
@@ -20,14 +20,14 @@ import {
...
 } from './SubscriptionConstants';
-
 import './SubscriptionsPage.scss';
 
 class SubscriptionsPage extends Component {
   componentDidMount() {
     this.props.resetTasks();
     this.props.loadSetting('content_disconnected');
     this.props.loadSubscriptions();
+    this.props.addModal({ id: MANAGE_MANIFEST_MODAL_ID });
```
Comment (MariaAga) [inline webpack/scenes/Subscriptions/SubscriptionsPage.js]: Why is this added here and in ManageManifestModal?

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js [hunk 6]
```diff
@@ -11,24 +12,23 @@ import { manifestExists } from '../SubscriptionHelpers';
 import { columns } from './ManifestHistoryTableSchema';
 import DeleteManifestModalText from './DeleteManifestModalText';
 
+export const MANAGE_MANIFEST_MODAL_ID = 'manageManifestModal';
+export const DELETE_MANIFEST_MODAL_ID = 'deleteManifestModal';
```
Comment (MariaAga) [inline webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: This should be in the consts file

File: webpack/scenes/Subscriptions/SubscriptionsPage.js (hunk 5 above)
Comment (jeremylenz) [reply webpack/scenes/Subscriptions/SubscriptionsPage.js]: @MariaAga Before I can use `setModalOpen` or `setModalClosed` the modal must be added.  I use those in both places, so that's why `addModal` is also in both.  The `addModal` action is idempotent (meaning I can send the same action twice and it won't hurt anything).  However, if I try to `setModalOpen` or `setModalClosed` and the modal doesn't exist, I throw an error.  So I think I was just doing it to be safe.

I guess I don't need to add it in `ManageManifestModal` since it should already be added in `SubscriptionsPage`.  I will remove it from `ManageManifestModal.js`

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js (hunk 6 above)
Comment (jeremylenz) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: I thought this made more sense here, in the same file that uses `ForemanModal`, because
* The constants file is customarily used for standardizing the names of Redux actions, but the ForemanModal ID is not an action name
* You could theoretically have a component that doesn't use any Redux actions but still has a ForemanModal ID, and is controlled from elsewhere
* Brings the value closer to where it is used.

I can also see the other side of the argument, which is "all constant string values should be organized in the same file."  But I'm partial to keeping it in the same file for the reasons above.  Thoughts?

Comment (MariaAga) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: I think the consts file is for all consts for example we also use it for `STATUS` which is not a Redux action but a value that we use.

> Brings the value closer to where it is used.

I saw that the ID was imported in another file so its not close to it :)

Comment (jeremylenz) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: @MariaAga ok fine, I'll move them :yum:

File: webpack/scenes/ModuleStreams/Details/__tests__/__snapshots__/ModuleStreamDetailsActions.test.js.snap [hunk 7]
```diff
@@ -134,11 +134,3 @@ Array [
...
 ]
 `;
-
-exports[`Module stream details actions should load module stream details on success: API get call 1`] = `
-Array [
-  Array [
-    "/module_streams/1",
-  ],
-]
-`;
```
Comment (MariaAga) [inline webpack/scenes/ModuleStreams/Details/__tests__/__snapshots__/ModuleStreamDetailsActions.test.js.snap]: Why is this and some other snapshots removed?

File: webpack/scenes/Subscriptions/SubscriptionConstants.js [hunk 8]
```diff
@@ -21,9 +21,6 @@ export const UPDATE_SUBSCRIPTION_COLUMNS = 'UPDATE_SUBSCRIPTION_COLUMNS';
 
 export const SUBSCRIPTIONS_UPDATE_SEARCH_QUERY = 'SUBSCRIPTIONS_UPDATE_SEARCH_QUERY';
 
-export const SUBSCRIPTIONS_OPEN_MANIFEST_MODAL = 'SUBSCRIPTIONS_OPEN_MANIFEST_MODAL';
-export const SUBSCRIPTIONS_CLOSE_MANIFEST_MODAL = 'SUBSCRIPTIONS_CLOSE_MANIFEST_MODAL';
```
Comment (MariaAga) [inline webpack/scenes/Subscriptions/SubscriptionConstants.js]: These are still in the reducer, should they be removed from it and its tests?

File: webpack/scenes/Subscriptions/__tests__/__snapshots__/SubscriptionsReducer.test.js.snap [hunk 9]
```diff
@@ -668,7 +668,7 @@ Object {
   "disconnected": false,
   "itemCount": 0,
   "loading": true,
-  "manifestModalOpened": false,
+  "manifestModalOpened": true,
```
Comment (MariaAga) [inline webpack/scenes/Subscriptions/__tests__/__snapshots__/SubscriptionsReducer.test.js.snap]: Why is this changed? 
`Subscriptions reducer should return the initial state 1` but initial state in the reducer is `  manifestModalOpened: false,`

File: webpack/scenes/ModuleStreams/Details/__tests__/__snapshots__/ModuleStreamDetailsActions.test.js.snap (hunk 7 above)
Comment (jeremylenz) [reply webpack/scenes/ModuleStreams/Details/__tests__/__snapshots__/ModuleStreamDetailsActions.test.js.snap]: Just some snapshot confusion.  hopefully fixing now

File: webpack/scenes/Subscriptions/SubscriptionConstants.js (hunk 8 above)
Comment (jeremylenz) [reply webpack/scenes/Subscriptions/SubscriptionConstants.js]: @MariaAga yep! Thanks

File: webpack/scenes/Subscriptions/__tests__/__snapshots__/SubscriptionsReducer.test.js.snap (hunk 9 above)
Comment (jeremylenz) [reply webpack/scenes/Subscriptions/__tests__/__snapshots__/SubscriptionsReducer.test.js.snap]: @MariaAga `manifestModalOpened` was used to keep track of the old modal, so it's no longer needed.   I removed it, thanks

File: webpack/scenes/Subscriptions/SubscriptionConstants.js [hunk 10]
```diff
@@ -21,12 +21,6 @@ export const UPDATE_SUBSCRIPTION_COLUMNS = 'UPDATE_SUBSCRIPTION_COLUMNS';
...
-
-export const SUBSCRIPTIONS_OPEN_DELETE_MODAL = 'SUBSCRIPTIONS_OPEN_DELETE_MODAL';
-export const SUBSCRIPTIONS_CLOSE_DELETE_MODAL = 'SUBSCRIPTIONS_CLOSE_DELETE_MODAL';
```
Comment (MariaAga) [inline webpack/scenes/Subscriptions/SubscriptionConstants.js]: Sorry didn't comment on these 2 as well
These are still in the reducer, should they be removed from it and its tests?

File: webpack/scenes/Subscriptions/index.js [hunk 11]
```diff
@@ -34,7 +33,6 @@ const mapStateToProps = (state) => {
     activePermissions: selectActivePermissions(state),
     tasks: selectSubscriptionsTasks(state),
     searchQuery: selectSearchQuery(state),
-    manifestModalOpened: selectManifestModalOpened(state),
     deleteModalOpened: selectDeleteModalOpened(state),
```
Comment (MariaAga) [inline webpack/scenes/Subscriptions/index.js]: Should be deleted as well as its not used

**Comment by jeremylenz** [comment]
@MariaAga Updated again.  let me know if you see any more leftovers.

File: webpack/scenes/Subscriptions/index.js (hunk 11 above)
Comment (MariaAga) [inline webpack/scenes/Subscriptions/index.js]: not resolved yet :)

Comment (jeremylenz) [reply webpack/scenes/Subscriptions/index.js]: Omg

**Comment by jeremylenz** [comment]
@MariaAga added back the missing deleteModal references

File: webpack/scenes/Subscriptions/Manifest/index.js [hunk 12]
```diff
@@ -11,15 +12,20 @@ import ManifestModal from './ManageManifestModal';
 
 import './Manifest.scss';
 
+export { MANAGE_MANIFEST_MODAL_ID } from './ManifestConstants';
```
Comment (MariaAga) [inline webpack/scenes/Subscriptions/Manifest/index.js]: why? all the components import it from `'./ManifestConstants'` anyway

Comment (jeremylenz) [reply webpack/scenes/Subscriptions/Manifest/index.js]: @MariaAga true, removing

File: webpack/components/Content/Details/__tests__/__snapshots__/ContentDetails.test.js.snap [hunk 13]
```diff
@@ -7,7 +7,7 @@ exports[`Content Details Info should render and contain appropriate components 1
     loadingText="Loading"
     timeout={300}
   >
-    <Uncontrolled(TabContainer)
+    <ForwardRef
```
Comment (MariaAga) [inline webpack/components/Content/Details/__tests__/__snapshots__/ContentDetails.test.js.snap]: I think this is unrelated to the PR as well as the other snap updates

Comment (jeremylenz) [reply webpack/components/Content/Details/__tests__/__snapshots__/ContentDetails.test.js.snap]: yes, hopefully this will be gone by the time this is rebased and ready to merge.

**Comment by jeremylenz** [comment]
Updated to not dispatch `addModal` action since ForemanModal takes care of it now.  Also rebased to latest master.

File: webpack/containers/Application/index.js [hunk 14]
```diff
@@ -17,6 +17,7 @@ export const organization = reducer;
 class Application extends Component {
   componentDidMount() {
     this.loadData();
+    console.log('Application didMount', this.props);
```
Comment (MariaAga) [inline webpack/containers/Application/index.js]: leftover?

Comment (jeremylenz) [reply webpack/containers/Application/index.js]: @MariaAga updated, thanks

**Comment by jeremylenz** [comment]
Katello test failures should go away once [Foreman #7105](https://github.com/theforeman/foreman/pull/7105) is merged

File: webpack/components/Content/Details/__tests__/__snapshots__/ContentDetails.test.js.snap (hunk 13 above)
Comment (jeremylenz) [reply webpack/components/Content/Details/__tests__/__snapshots__/ContentDetails.test.js.snap]: update: gone :)

**Comment by jeremylenz** [comment]
Rebased to latest `master`

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js [hunk 15]
```diff
@@ -150,166 +147,147 @@ class ManageManifestModal extends Component {
...
-                              </Tooltip>
-                            }
-                            placement="bottom"
-                            trigger={['hover', 'focus']}
-                            rootClose={false}
-                          >
-                            <span>{__('USMA')}</span>
-                          </OverlayTrigger>
+      <ForemanModal id={MANAGE_MANIFEST_MODAL_ID} title={__('Manage Manifest')}>
+        <ForemanModal.Header />
```
Comment (MariaAga) [inline webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: Is this still needed after the change in foreman core?

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js [hunk 16]
```diff
@@ -150,166 +147,147 @@ class ManageManifestModal extends Component {
...
+                              tooltipId="delete-manifest-button-tooltip"
+                              tooltipText={this.disabledTooltipText()}
+                              tooltipPlacement="top"
+
+                            />
+                          </React.Fragment>
+                          }
+                        </div>
+                        <ForemanModal title={__('Confirm delete manifest')} id={DELETE_MANIFEST_MODAL_ID}>
+                          <ForemanModal.Header />
```
Comment (MariaAga) [inline webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: Same, is this needed?

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js (hunk 15 above)
Comment (jeremylenz) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: @MariaAga no, updating

**Comment by jeremylenz** [comment]
Rebased

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js [hunk 17]
```diff
@@ -150,166 +147,145 @@ class ManageManifestModal extends Component {
...
+                          </React.Fragment>
+                          }
+                        </div>
+                        <ForemanModal title={__('Confirm delete manifest')} id={DELETE_MANIFEST_MODAL_ID}>
+                          <DeleteManifestModalText />
+                          <ForemanModal.Footer>
+                            <Button bsStyle="default" onClick={this.hideDeleteManifestModal}>
+                              {__('Cancel')}
+                            </Button>
+                            <Button bsStyle="danger" onClick={() => this.deleteManifest()}>
```
Comment (MariaAga) [inline webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: Can this be ` onClick= {this.deleteManifest}` ?

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js [hunk 18]
```diff
@@ -150,166 +147,145 @@ class ManageManifestModal extends Component {
...
+              <Table
+                rows={manifestHistory.results}
+                columns={columns}
+                emptyState={emptyStateData()}
+              />
+            </LoadingState>
+          </Tab>
+        </Tabs>
+        <ForemanModal.Footer>
           <Button bsStyle="primary" onClick={this.hideModal}>
```
Comment (MariaAga) [inline webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: I'm getting an error here when trying to close the modal:
`ForemanModalActions.js:35 Uncaught Error: SET_MODAL_CLOSED error: Modal with id 'deleteManifestModal' does not exist`

Comment (jeremylenz) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: @MariaAga I'm not seeing this error locally, do you remember what steps you were going through to reproduce it?

Comment (MariaAga) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: https://github.com/Katello/katello/blob/db28f63630d19f6371a1122f6441d2afce39db9d/webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js#L110

You can change `showSubscriptionManifest` to false to reproduce it

Comment (jeremylenz) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: @MariaAga Thanks, updated.  I may do a PR in Foreman to not throw this error anyway.  It feels like it's too much, and since we use constants for modal ID anyway, I feel like that's enough of a guard against using the wrong modal ID when managing your modals.

Comment (MariaAga) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: Great!

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js (hunk 17 above)
Comment (jeremylenz) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: @MariaAga updated
